src/main/java/bio/terra/service/resourcemanagement/azure/AzureMonitoringService.java

```diff
@@ -0,0 +1,622 @@
+   * @param storageAccount The Storage Account being logged
+   * @param workspaceId ID of the Log Analytics workspace that the diagnostic setting is associated
+   *     with
+   * @return A string representing the id of the storage account diagnostic setting that was created
+   */
+  public String createDiagnosticSetting(
+      BillingProfileModel profileModel,
+      AzureStorageAccountResource storageAccount,
+      String workspaceId) {
+    AzureResourceManager client = resourceConfiguration.getClient(profileModel.getSubscriptionId());
+
+    logger.info(
+        "Creating new Log Analytics Workspace Diagnostic settings for Storage Account {}",
+        storageAccount.getStorageAccountId());
+
+    // Obtain all categories that can be tracked
+    List<DiagnosticSettingsCategory> logCategories =
+        client
+            .diagnosticSettings()
+            .listCategoriesByResource(getStorageAccountLoggingResourceId(storageAccount))
+            .stream()
+            .toList();
+
+    return client
+        .diagnosticSettings()
+        .define(storageAccount.getName())
+        .withResource(getStorageAccountLoggingResourceId(storageAccount))
+        .withLogAnalytics(workspaceId)
+        .withLogsAndMetrics(logCategories, METRIC_GRANULARITY, LOG_DATA_RETENTION_DAYS)
+        .create()
+        .id();
+  }
+
+  /**
+   * Delete an existing Log Analytics diagnostic setting
+   *
+   * @param profileModel The billing profile for the dataset or snapshot being logged
+   * @param id The Azure id of the Log Analytics diagnostic setting to delete
+   */
+  public void deleteDiagnosticSetting(BillingProfileModel profileModel, String id) {
+    AzureResourceManager client = resourceConfiguration.getClient(profileModel.getSubscriptionId());
+
+    logger.info("Deleting Log Analytics Workspace Diagnostic settings {}", id);
+
+    client.diagnosticSettings().deleteById(id);
+  }
+
+  /**
+   * Retrieve an existing data export rule for the given Log Analytics workspace
+   *
+   * @param profileModel The billing profile for the dataset or snapshot being logged
+   * @param storageAccount The Storage Account being logged
+   * @return A {@link DataExport} object representing the data export rule or null if it does not
+   *     exist
+   */
+  public DataExport getDataExportRule(
+      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
+
+    LogAnalyticsManager client =
+        resourceConfiguration.getLogAnalyticsManagerClient(
+            resourceConfiguration.getCredentials().getHomeTenantId(),
+            profileModel.getSubscriptionId());
+
+    try {
+      DataExport dataExport =
+          client
+              .dataExports()
+              .get(
+                  storageAccount.getApplicationResource().getAzureResourceGroupName(),
+                  storageAccount.getName(),
+                  storageAccount.getName());
+      logger.debug("Found Log Analytics Workspace data export rule");
```

Comment:
This uses `debug()` but other similar code in this file use `info()` such as `getLogAnalyticsWorkspace()`. Is there a need to use different logging levels for this message vs the others?